packages/form-field/src/FormField.spec.tsx

```diff
@@ -184,34 +197,145 @@ describe('packages/form-field', () => {
+            </FormFieldInputContainer>
+          </FormField>,
+        );
+        const input = getByTestId('input');
+        expect(input).toHaveAttribute('aria-labelledby', ariaLabelledby);
+      });
+
+      test('when aria-labelledby and aria-label are defined, input does not set the aria-label', () => {
+        const { getByTestId } = render(
+          <FormField
+            aria-labelledby={ariaLabelledby}
+            aria-label={ariaLabel}
+            data-testid="form-field"
+          >
+            <FormFieldInputContainer>
+              <div data-testid="input" />
+            </FormFieldInputContainer>
+          </FormField>,
+        );
+        const input = getByTestId('input');
+        expect(input).toHaveAttribute('aria-labelledby', ariaLabelledby);
+        expect(input).not.toHaveAttribute('aria-label', ariaLabel);
+      });
+
+      test('when aria-labelledby is not defined and aria-label is defined, input sets aria-label attribute to custom prop', () => {
+        const { getByTestId } = render(
+          <FormField aria-label={ariaLabel} data-testid="form-field">
+            <FormFieldInputContainer>
+              <div data-testid="input" />
+            </FormFieldInputContainer>
+          </FormField>,
+        );
+        const input = getByTestId('input');
+        expect(input).toHaveAttribute('aria-label', ariaLabel);
+      });
+    });
+  });
+
+  describe(`${FormFieldState.Valid} state`, () => {
+    test('Success icon and message are not shown by default', () => {
+      const { container, queryByText } = render(
+        <FormField
+          label="Label"
+          successMessage="Success"
+          data-testid="form-field"
+        >
           <FormFieldInputContainer>
             <div data-testid="input" />
           </FormFieldInputContainer>
         </FormField>,
       );
-      const input = getByTestId('input');
-      expect(input).toHaveAttribute('aria-label', 'Label');
+      const successIcon = container.querySelector('svg');
+      const successText = queryByText('Success');
+      expect(successIcon).toBeNull();
+      expect(successText).toBeNull();
     });
 
-    test('when aria-labelledby is provided, input has that aria-labelledby', () => {
-      const { getByTestId } = render(
-        <FormField aria-labelledby="label-123" data-testid="form-field">
+    test(`Success icon and message appear when state is ${FormFieldState.Valid}`, () => {
+      const { container, queryByText } = render(
+        <FormField
+          label="Label"
+          successMessage="Success"
+          state={FormFieldState.Valid}
+          data-testid="form-field"
+        >
           <FormFieldInputContainer>
             <div data-testid="input" />
           </FormFieldInputContainer>
         </FormField>,
       );
-      const input = getByTestId('input');
-      expect(input).toHaveAttribute('aria-labelledby', 'label-123');
+      const successIcon = container.querySelector('svg');
```

Comment:
To check that the correct icon renders, you could search for the `aria-label` or `title` of the SVG.

Reply:
updated!